engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -262,6 +256,29 @@ private ConstrainedField findPropertyMetaData(Field field) {
 		);
 	}
 
+	private UnwrapMode unwrapMode(Field field, boolean typeArgumentAnnotated) {
```

Comment:
Shouldn't something similar also be done for (getter-defined) properties and method-level constraints?